plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -182,6 +183,83 @@ export class SonataFlowService {
     return undefined;
   }
 
+  public async fetchWorkflowOverviews(): Promise<
+    WorkflowOverview[] | undefined
+  > {
+    try {
+      const response = await executeWithRetry(() =>
+        fetch(`${this.url}/management/processes`),
+      );
+
+      if (response.ok) {
+        const workflowIds = (await response.json()) as string[];
+        if (!workflowIds?.length) {
+          return [];
+        }
+        const items = await Promise.all(
+          workflowIds.map(async (workflowId: string) => {
+            const definition = await this.fetchWorkflowDefinition(workflowId);
+            if (!definition) {
+              return undefined;
+            }
+
+            const graphQlQuery = `{ ProcessInstances(where: {processId: {equal: "${definition.id}" } } ) { processName, state, start, lastUpdate, end } }`;
+            let processInstances: ProcessInstance[] = [];
+            try {
+              const response = await executeWithRetry(() =>
+                fetch(`${this.url}/graphql`, {
+                  method: 'POST',
+                  body: JSON.stringify({ query: graphQlQuery }),
+                  headers: { 'content-type': 'application/json' },
+                }),
+              );
+
+              if (response.ok) {
+                const json = await response.json();
+                processInstances = json.data.ProcessInstances;
+              }
```

Comment:
no else block? (I see the same is in the existing function fetchProcessInstances)

Reply:
Do you suggest returning the data retrieval status to the frontend instead of the hardcoded 500 status?

Reply:
04f19541b453bbd281be4ee932ca7d3ab6af7d31